src/System/Wlog/PureLogging.hs

```diff
@@ -126,9 +127,18 @@ launchNamedPureLog hoist' (NamedPureLogger action) = do
     res <$ dispatchEvents logs
 
 -- | Similar to 'runNamedPureLog', but using provided logger name.
-usingPureLoggerName :: Functor m
-                    => LoggerName
-                    -> NamedPureLogger m a
-                    -> m (a, [LogEvent])
-usingPureLoggerName name (NamedPureLogger action) =
+usingNamedPureLogger :: Functor m
+                     => LoggerName
+                     -> NamedPureLogger m a
+                     -> m (a, [LogEvent])
+usingNamedPureLogger name (NamedPureLogger action) =
     usingLoggerName name $ runPureLog action
+
+-- | Perform pure-logging computation, log its events
+-- and return the result of the computation
+logPureAction :: WithLogger m => NamedPureLogger m a -> m a
+logPureAction namedPureLogger = do
+    loggerName    <- askLoggerName
+    (a, logEvnts) <- usingNamedPureLogger loggerName namedPureLogger
+    logEvents logEvnts
+    pure a
```

Comment:
This can be written in shorter form `a <$ dispatchEvents eventsBacklog`.

I also think that it's better to use `dispatchEvents` here instead of `logEvents` because you already have `LogEvent`s with proper `LoggerName`. You don't need to use `askLoggerName` (inside `logEvents`) second time.